packages/ens-utils/src/namekitregistrar.ts

```diff
@@ -0,0 +1,51 @@
+import { buildAddress } from "./address";
+import { MAINNET } from "./chain";
+import { buildContractRef, ContractRef } from "./contract";
+import { Currency } from "./currency";
+import { charCount, ENSName } from "./ensname";
+import { EthRegistrar } from "./ethregistrar";
+import { buildPrice, Price } from "./price";
+import {
+  RegistrarCharge,
+  RegistrarChargeType,
+  RegistrarServiceFee,
+} from "./registrar";
+
+// example of how to modify a pricing policy to add a service fee to the base registration rate
+export class NameKitRegistrar extends EthRegistrar {
+  public constructor(registrarContract: ContractRef) {
+    super(registrarContract);
+  }
+
+  protected getAnnualCharges(name: ENSName): RegistrarCharge[] {
+    const baseCharges = super.getAnnualCharges(name);
+    const subname = this.getValidatedSubname(name);
+    const subnameLength = charCount(subname.name);
+    let annualServiceFee: Price;
+    if (subnameLength === 3 || subnameLength == 4) {
+      annualServiceFee = buildPrice(999n, Currency.Usd); // $9.99 USD
+    } else {
+      annualServiceFee = buildPrice(99n, Currency.Usd); // $0.99 USD
+    }
+
+    const serviceFee: RegistrarServiceFee = {
+      type: RegistrarChargeType.ServiceFee,
+      price: annualServiceFee,
+      reason: "Example NameKit convenience fee.", // TODO: update this
```

Comment:
@lightwalker-eth need any help with ideation?

Reply:
@FrancoAguzzi Right now my mindset is to optimize for merging this PR sooner rather than later. Thanks.